oneview_redfish_toolkit/api/scmb.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2017) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import json
+import logging
+import os
+import ssl
+import sys
+
+from hpOneView.exceptions import HPOneViewException
+import pika
+from pika.credentials import ExternalCredentials
+
+
+from oneview_redfish_toolkit import util
+
+
+ONEVIEW_CA = "certs/oneview_ca.pem"
+SCMB_CERT = "certs/oneview_scmb.pem"
+SCMB_KEY = "certs/oneview_scmb.key"
+
+
+def check_cert_exist():
+    if not os.path.isfile(ONEVIEW_CA):
+        return False
+    if not os.path.isfile(SCMB_CERT):
+        return False
+    if not os.path.isfile(SCMB_KEY):
+        return False
+    return True
+
+
+def get_cert():
+    # Get CA
+    cert = util.ov_client.certificate_authority.get()
+    with open(util.config['scmb']['oneview_ca'], 'w+') as f:
+        f.write(cert)
+    # Generate scmb Cert:
+    try:
+        cert_info = {
+            "commonName": "default",
+            "type": "RabbitMqClientCertV2"
+        }
+        util.ov_client.certificate_rabbitmq.generate(cert_info)
+    except HPOneViewException as e:
+        print(dir(e.oneview_response))
+        # Cert with that commonName already exists. We are going to get it
+        if e.oneview_response["errorCode"] == 'RABBITMQ_CLIENTCERT_CONFLICT':
+            logging.info('Certs already exists in oneview')
+        else:
+            # Another error is not expected, we raise.
+            raise
+    # Get the scmb certs key pair
+    certs = util.ov_client.certificate_rabbitmq.get_key_pair(
+        'default')
+    # Save cert
+    with open(util.config['scmb']['scmb_cert'], 'w+') as f:
+        f.write(certs['base64SSLCertData'])
+    # Save key
+    with open(util.config['scmb']['scmb_key'], 'w+') as f:
+        f.write(certs['base64SSLKeyData'])
+
+
+def scmb_connect():
+    scmb_server = util.config['oneview_config']['ip']
+
+    # Setup our ssl options
+    ssl_options = ({'ca_certs': ONEVIEW_CA,
+                    'certfile': SCMB_CERT,
+                    'keyfile': SCMB_KEY,
+                    'cert_reqs': ssl.CERT_REQUIRED,
+                    'server_side': False})
+
+    scmb_connection = pika.BlockingConnection(
+        pika.ConnectionParameters(
+            scmb_server, 5671, credentials=ExternalCredentials(),
+            ssl=True, ssl_options=ssl_options))
+    return scmb_connection
+
+
+def test_cert():
```

Comment:
This function has a weird name. Should this be called from listen_scmb and not called from the app.py?

Reply:
It's goal is to make sure that the certs I have work with OneView's rabbitMQ. That's why 'test_cert'. It establishes a connection to rabbitMQ and tries to bind to queue. I was not sure that only calling connect would be enough so i decided to go all the way into queue listening to make sure the connection with the cert works fine. The reason I'm calling this from the app is because it's a test. I wish to call listen_scmb (which very soon will be a bunch of threads) when I'm positive that the connection works, I didn't what to do is several times when I'm in a loop to launch threads. If you still find the function name weird now that you know it's purpose  please feel free to to suggest a better name. If you still think that the test location taking in consideration my arguments let me know, I can change that, but still think it's not the best place to put it.

Reply:
I prefer that only unit/function tests start with `test_`; a better option would be `is_cert_working_with_scmb` based on the usage of it below...

Reply:
What about the discussion if we should test or not before launching threads?!?!?
The name will be fixed

Reply:
I missed this... but I think that we can fail the whole toolkit gracefully the way it is implemented right now. We could wait and get the result of opening the new thread and then fail but it would be more troublesome.

However, @danielreed, if your idea is to leave the toolkit endpoints up even if the SCMB could not be opened... then I need to think a bit more about this scenario!